scipy/special/special/mdspan.h

```diff
@@ -0,0 +1,5268 @@
+// Created by SciPy developers in 2024 running make_single_header.py
+// See https://github.com/kokkos/mdspan
+
+#ifndef _MDSPAN_SINGLE_HEADER_INCLUDE_GUARD_
+#define _MDSPAN_SINGLE_HEADER_INCLUDE_GUARD_
+
+//BEGIN_FILE_INCLUDE: scipy/special/special/mdspan/include/experimental/mdspan
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
```

Comment:
This license looks like a problem. I'm not super familiar with the LLVM exception part, but Apache 2.0 isn't a compatible license.

Reply:
Right, thanks for pointing that out @rgommers. I leave the license stuff to you. But sounds like the following might acceptable?

The primary motivation for us right now is to turn the remaining gufunc-like functions like the Legendre polynomials, etc into a form that allows us to cleanly split off special soon as we can. It just so happens an earlier version of mdspan was released under the BSD license: https://github.com/ORNL/cpp-proposals-pub/blob/master/P0009/reference-implementation/include/experimental/bits/mdspan.hpp

I guess there are no issues with that? And, in the meantime, SciPy can have general discussions about the Apache license if anyone wants to do that.

Reply:
Yes, if the earlier BSD-3 version is usable, that's probably still a lot better than reinventing the wheel here. I'll have a closer look after the weekend.